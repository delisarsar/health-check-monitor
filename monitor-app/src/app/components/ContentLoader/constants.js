import React from "react";
import { FormattedMessage } from "react-intl";

export const GENERIC_APP_ERROR = (
  <FormattedMessage
    id="app.error"
    defaultMessage={"An error occured. Please try again"}
  />
);
