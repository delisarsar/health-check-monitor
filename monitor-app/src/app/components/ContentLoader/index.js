import React from "react";
import PropTypes from "prop-types";
import CircularProgress from "@material-ui/core/CircularProgress";
import { Typography } from "@material-ui/core";

const ContentLoader = (props) => {
  const { loading, error, children } = props;
  if (loading) {
    return <CircularProgress data-testid="progressLoader" />;
  }
  if (error) {
    return <Typography data-testid="errorMessage">Error happened</Typography>;
  }
  return children;
};

export default ContentLoader;

ContentLoader.propTypes = {
  children: PropTypes.node,
  loading: PropTypes.bool,
  error: PropTypes.object,
};
