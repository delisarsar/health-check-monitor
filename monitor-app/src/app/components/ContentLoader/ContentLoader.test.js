import React from "react";
import { render } from "@testing-library/react";
import ContentLoader from ".";
import { Typography } from "@material-ui/core";

test("renders correctly the children", () => {
  const { getByText } = render(
    <ContentLoader>
      <Typography>Hello</Typography>
    </ContentLoader>
  );
  expect(getByText("Hello")).toBeInTheDocument();
});

test("renders a circular progress when loading is true", () => {
  const { getByTestId } = render(<ContentLoader loading={true} />);
  expect(getByTestId("progressLoader")).toBeInTheDocument();
});

test("renders an error message when it's true", () => {
  const { getByTestId } = render(<ContentLoader error={true} />);
  expect(getByTestId("errorMessage")).toBeInTheDocument();
});
