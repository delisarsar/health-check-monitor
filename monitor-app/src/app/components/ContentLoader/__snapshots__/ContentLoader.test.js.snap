// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a circular progress when loading is true 1`] = `ShallowWrapper {}`;

exports[`renders correctly the children 1`] = `ShallowWrapper {}`;
